### PR TITLE
Documentation of Single APK Approach Configuration

### LINKED ISSUE #1284: [Documentation] - Create Documentation of Single APK Approach Configuration
Add documentation that will allow us to easily understand
1. The structure of SAA app configs 
2. How the config is interpreted and used in-app 
3. How to add production and development/local configs
4. How to use local configurations in development mode. All you have to do is append the /debug suffix to an appId that is available in the assets directory. E.g. usage: quest/debug, anc/debug

Acceptance Criteria/Output

1. Create a markdown document that can be saved in the docs for FHIR Core

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1284 

To view the documentation in a non-raw form, open this link
https://github.com/opensrp/fhircore/blob/documenting-configurable-app/docs/configuration.md

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: docs/configuration.md
Comment (FikriMilano) [reply docs/configuration.md:618]: Thanks for the feedback.
Sure I'll do that

Comment (FikriMilano) [reply docs/configuration.md:618]: By this complex expression `focus.subject eq Organization/#organization`
Can I assume the API request link will look like this?

https://fhir.labs.smartregister.org/fhir/[SOME_RESOURCE]?focus.subject=Organization/101

Correct me if I'm wrong.

Comment (maimoonak) [reply docs/configuration.md:618]: No. The tables looks good and example is also fine. Just wanted to mention that there could be numerous way someone would want to sync their data. And since its FHIR way of we need to point to FHIR docs on how to write a sync config expression or code i.e. how do some know that code can be patient.name or any other value. And also the expression can be simple or complex, so how do someone learn to write complex expressions. So If we can also add these FHIR references and explanation on possible values in sync config that would be very helpful.

The complete expression is "patient.organization eq #organization or focus.subject eq Organization/#organization".
So link might look like 
[https://fhir.labs.smartregister.org/fhir/Observation?_filter=patient.organization eq 361 or focus.subject eq Organization/361](https://fhir.labs.smartregister.org/fhir/Observation?_filter=patient.organization%20eq%20361%20or%20focus.subject%20eq%20Organization/361)

Comment (FikriMilano) [reply docs/configuration.md:618]: I want to understand more about this case of complex expression:
`patient.organization eq #organization or focus.subject eq Organization/#organization`

`patient` is a SearchParameter from Observer.
It filters based on the path of `Observer.subject` it gets the subject inside the Observer.

Then how about the `patient.organization`?
Does that means it filters based on the Observer's Patient subject via the `patient` SearchParameter,
Then,
It gets the Patient's Organization via the `organization` SearchParameter? which Patient resource has.
https://build.fhir.org/patient.html#search

Is that right?

Comment (maimoonak) [reply docs/configuration.md:618]: yes. this is called chained parameters. https://hl7.org/fhir/search.html#chaining

**Review by maimoonak — APPROVED** [review]

**Review by pld — CHANGES_REQUESTED** [review]
I want to review text before merge, blocking merge for the moment

**Review by pld — APPROVED** [review]

**Pull request merged**
